src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs

```diff
@@ -911,24 +867,38 @@ private void InitializeFSM()
                     case OldestChangedBuffer.OldestChanged oldestChanged when e.StateData is YoungerData youngerData:
                     {
                         _oldestChangedReceived = true;
-                        if (oldestChanged.Oldest != null && oldestChanged.Oldest.Equals(_selfUniqueAddress))
+                        if (oldestChanged.NewOldest != null && oldestChanged.NewOldest.Equals(_selfUniqueAddress))
```

Comment:
This and a corresponding change in the `OldestChangedBuffer` are the two big fixes in this PR, aside from dealing with some of the minor mutability issues.

## The Problem

When we receive a `MemberRemoved` for an older node that was previously hosting a cluster singleton, we intentionally delay processing of that `MemberRemoved` event via the `DelayedMemberRemoved` mechanism. This is designed to ensure that in the event that there is a split brain and the SBR was responsible for removing that old node, that node still gets a chance to shut itself down and die cleanly before we take over. This is especially important for singletons using Akka.Persistence, so we don't have two competing actors using the same persistence id and database. So, we still assume that node is the oldest up to ~20s after it's removal.

This is normally not an issue - when we attempt to send hand-overs to that node they'll time out eventually and we'll just recreate the singleton in a matter of 10-20 seconds typically.

However, where this design creates a problem is during rolling updates in a managed environment like Kubernetes, where restarts can happen really quickly.

Imagine this scenario:

* Cluster of `node1` (oldest), `node2`, `node3` - all have the right `role` to host the singleton.
* `node1` is replaced first by K8s - it's terminated via CLR process exit; uses `CoordinatedShutdown` to try to exit gracefully as best as it can.
* `node2` sees that `node1` is exiting and does the correct thing - messages it for a handover and becomes the new singleton host and is now the oldest node.
* Kubernetes restarts `node1`, which joins the cluster under a new `UniqueMemberAddress` but with the same `Address` as before.
* Kubernetes then takes down `node2` - per the update process.
* If all of the prior steps happen in under 20 seconds, `node3` will be notified that it is becoming the oldest node and it will start messaging `node1`, which is ACTUALLY YOUNGER THAN IT, to do the hand-over. `node1` will ACK that hand-over right away and we'll have two separate cluster singletons active at the same time, until `node2` completely exits. This issue is one of the two bugs that caused https://github.com/akkadotnet/akka.net/issues/6973 - the member ordering problem that we fixed on https://github.com/akkadotnet/akka.net/pull/7297 being the other.


So to fix this issue, we can either speed up member removals or do something even simpler - just include the address of the previous "oldest" node in the `OldestChanged` message and put that one at the front of the list in our `Younger` and `BecomingOldest` states. That way, we're always messaging the most recent home of the singleton using live data from the `OldestChangedBuffer`. I've run this experiment thousands of times in our test lab and can confirm that it eliminated the issue completely.